### PR TITLE
New Quarkus platform composition and config

### PR DESCRIPTION
**Pull request by aloubyansky**
FYI @gsmet @maxandersen 

The README still needs to be updated. The `quarkus-maven-plugin` is now re-published with platform release under the platform's groupId and version. Not sure I can anything here with the Gradle one. Otherwise, it's pretty much complete.

There are two commits:
* the platform config;
* the generated platform project.

The platform gets re-generated on every build. The reason it is checked in is to clearly see how the change in the platform config (the root `pom.xml`) affects the generated platform project.

### CODE REVIEW

**Comment by aloubyansky** [comment]
fyi @ppalaga

**Review by gsmet — COMMENTED** [review]
Added a small question.

File: generated-platform-project/quarkus-blaze-persistence/descriptor/pom.xml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+              <goal>generate-platform-descriptor</goal>
+            </goals>
+          </execution>
+        </executions>
+        <configuration>
+          <bomArtifactId>quarkus-blaze-persistence-bom</bomArtifactId>
+          <quarkusCoreVersion>2.0.0.Alpha3</quarkusCoreVersion>
+          <platformRelease>
+            <platformKey>io.quarkus.platform</platformKey>
+            <stream>999-SNAPSHOT</stream>
```
[the remark this answers is not in the document]
Comment (aloubyansky) [reply generated-platform-project/quarkus-blaze-persistence/descriptor/pom.xml:40]: Yes, of course, this is generated. This value can be produced in two ways:
1) Deduced from the platform version automatically, i.e. by default the stream currently is `major.minor` combination, so if you set the platform version to `2.0.0`, the stream here will be `2.0`. If you then release `2.0.1`, it's still going to be `2.0` of course.
2) there is a comment in the root pom under `releaseConfig` where the stream and version can be configured explicitly.

This is the initial version and may need further refinement.

File: generated-platform-project/quarkus-blaze-persistence/descriptor/pom.xml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+              <member>io.quarkus.platform:quarkus-bom-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT</member>
+              <member>io.quarkus.platform:quarkus-cassandra-bom-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT</member>
+              <member>io.quarkus.platform:quarkus-hazelcast-client-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT</member>
+              <member>io.quarkus.platform:quarkus-optaplanner-bom-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT</member>
+              <member>io.quarkus.platform:quarkus-qpid-jms-bom-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT</member>
+              <member>io.quarkus.platform:quarkus-blaze-persistence-bom-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT</member>
+              <member>io.quarkus.platform:quarkus-debezium-outbox-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT</member>
+              <member>io.quarkus.platform:quarkus-camel-bom-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT</member>
+              <member>io.quarkus.platform:quarkus-kogito-bom-quarkus-platform-descriptor:999-SNAPSHOT:json:999-SNAPSHOT</member>
```
[the remark this answers is not in the document]
Comment (aloubyansky) [reply generated-platform-project/quarkus-blaze-persistence/descriptor/pom.xml:51]: These versions match the version set in the root project pom.

**Review by ppalaga — APPROVED** [review]
Great piece of work, thanks @aloubyansky ! I have filed a couple of minor wishes https://github.com/quarkusio/quarkus-platform-bom-generator/issues but none of those is critical.

File: generated-platform-project/quarkus-blaze-persistence/descriptor/pom.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (gsmet) [reply generated-platform-project/quarkus-blaze-persistence/descriptor/pom.xml:40]: Yeah, what I'm saying is that somehow, we will need to update all this in the release process and get things committed before tagging.

**Comment by ppalaga** [comment]
:tada:

**Pull request merged**